Automattic/wp-calypso
# State: Move Jetpack module notices to the respective actions

#### Changes proposed in this Pull Request

As we've previously discussed in other PRs, the notices middleware isn't as useful as it looks like. It adds some weight that is loaded on all pages on the main entry point but is still used far too seldom to make sense to always be loaded.

This PR removes the Jetpack module notices away from the notices middleware, and into the Jetpack module (de)activation action creators where they belong. Note that this only applies to Jetpack modules, simple sites shouldn't be affected.

This PR also uses the opportunity to clean up several things:
* Removing the per-module notices as they're not used (they only exist for infinite scroll, and are no longer used there since that module no longer displays a toggle for Jetpack).
* Unifying success and failure cases and reduce code repetition.
* Declaring `noticeSettings` only once.

#### Testing instructions

* Checkout this branch.
* Go to `/settings/writing/:site` where `:site` is a Jetpack site.
* Play with module toggles like WP.com toolbar and verify you still get notices the same way as in production.
* Test on a WP.com simple site and verify those aren't affected either.
